### PR TITLE
[BUGFIX] Follow up to (#68) - fixed crashing on disconnect

### PR DESCRIPTION
**Pull request by cartertinney**
* Fixed a bug that caused a crash upon disconnect
* Re-enabled the clearing of method requests from inbox upon disconnect now that crashing is fixed
* Added tests for client handlers sent to the transport
* Added tests for any parts of the direct methods code that is run in a common code path shared with other features
* Fixed an additional direct methods bug that would have orphaned inboxes from the inbox manager.
* Made minor sample adjustments to align output with similar samples.

### CODE REVIEW

File: azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/inbox_manager.py [hunk 1]
```diff
@@ -77,10 +77,11 @@ def get_method_request_inbox(self, method_name=None):
         return inbox
 
     def clear_all_method_requests(self):
```
Comment (pierreca) [inline azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/inbox_manager.py]: >clear_all_method_requests [](start = 8, length = 25)

nit: just to make this clear i'd suggest renaming this to "delete_method_inboxes". deleting inboxes implies clearing the method requests, the other way doesn't seem true. not a blocker.

File: azure-iot-hub-devicesdk/samples/send_to_output.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 module_client.connect()
 
 # send 5 messages with a 1 second pause between each message
-for i in range(0, 5):
+for i in range(1, 6):
```
Comment (pierreca) [inline azure-iot-hub-devicesdk/samples/send_to_output.py]: >(1, 6): [](start = 14, length = 7)

this seems unrelated to the main objective of the PR (not a blocker, just making sure this is intentional)

File: azure-iot-hub-devicesdk/tests/test_sync_clients.py [hunk 3]
```diff
@@ -202,6 +229,13 @@ class TestDeviceClient(ClientSharedTests):
     def client(self, transport):
         return DeviceClient(transport)
 
+    def test_instantiation_sets_on_c2d_message_recieved_handler_in_transport(self, client):
```
Comment (pierreca) [inline azure-iot-hub-devicesdk/tests/test_sync_clients.py]: >recieved [](start = 47, length = 8)

nit: typo (received)

File: azure-iot-hub-devicesdk/tests/aio/test_async_clients.py [hunk 4]
```diff
@@ -120,6 +142,13 @@ class TestModuleClient(ClientSharedTests):
     def client(self, transport):
         return ModuleClient(transport)
 
+    async def test_instantiation_sets_on_input_message_recieved_handler_in_transport(self, client):
```
Comment (pierreca) [inline azure-iot-hub-devicesdk/tests/aio/test_async_clients.py]: >recieve [](start = 55, length = 7)

same typo here

File: azure-iot-hub-devicesdk/samples/send_to_output.py (hunk 2 above)
Comment (cartertinney) [reply azure-iot-hub-devicesdk/samples/send_to_output.py]: "Made minor sample adjustments to align output with similar samples."

This sample was the one that was causing problems, and in taking a close look at it, I realized the output wasn't alligned with most other samples, so I fixed it too.

---
In reply to: [268366269](https://github.com/Azure/azure-iot-sdk-python-preview/pull/69#discussion_r268366269) [](ancestors = 268366269)

File: azure-iot-hub-devicesdk/tests/aio/test_async_clients.py (hunk 4 above)
Comment (cartertinney) [reply azure-iot-hub-devicesdk/tests/aio/test_async_clients.py]: I hate this spelling so much

---
In reply to: [268366634](https://github.com/Azure/azure-iot-sdk-python-preview/pull/69#discussion_r268366634) [](ancestors = 268366634)
